src/libmongoc/tests/test-mongoc-sample-commands.c

```diff
@@ -3703,13 +3703,146 @@ _test_sample_versioned_api_example_4 (void)
    mongoc_server_api_destroy (server_api);
 }
 
+static int64_t iso_to_unix (const char* iso_str) {
+   /* TODO (CDRIVER-2945) there is no convenient helper for converting ISO8601
+    * strings to Unix timestamps. This is not shown in the example. */
+   return 1628330345;
+}
+
+static void _test_sample_versioned_api_example_5_6_7_8 (void) {
+#define N_DOCS 8
+   mongoc_client_t *client = NULL;
+   mongoc_server_api_t *server_api = NULL;
+   mongoc_server_api_version_t server_api_version;
+   bool ok;
+   bson_error_t error;
+   mongoc_database_t *db;
+   mongoc_collection_t *sales;
+   bson_t *docs[N_DOCS];
+   int i;
+   bson_t reply;
+   bson_t *cmd;
+   int64_t count;
+   bson_t *filter;
+
+   client = get_client ();
+   mongoc_client_set_error_api (client, MONGOC_ERROR_API_VERSION_2);
+   mongoc_server_api_version_from_string ("1", &server_api_version);
+   server_api = mongoc_server_api_new (server_api_version);
+   mongoc_server_api_strict (server_api, true);
+   ok = mongoc_client_set_server_api (client, server_api, &error);
+   ASSERT_OR_PRINT (ok, error);
+   db = mongoc_client_get_database (client, "db");
+   sales = mongoc_database_get_collection (db, "sales");
+   /* Drop db.sales in case the collection exists. */
+   ok = mongoc_collection_drop (sales, &error);
+   if (!ok && NULL == strstr (error.message, "ns not found")) {
+      /* Ignore an "ns not found" error on dropping the collection in case the
+       * namespace does not exist. */
+      ASSERT_OR_PRINT (ok, error);
+   }
+
+   /* Start Versioned API Example 5 */
+   docs[0] = BCON_NEW ("_id", BCON_INT32 (1),
+                       "item", "abc",
+                       "price", BCON_INT32 (10),
+                       "quantity", BCON_INT32 (2),
+                       "date", BCON_DATE_TIME (iso_to_unix ("2021-01-01T08:00:00Z")));
```

Comment:
The `iso_to_unix` function is a no-op. It is not present in the documentation example (i.e. not within the `Start` / `End` comments. The intent is to convey that there should be some conversion to the unix timestamp so this can be expressed as a BSON datetime type.